platform/broadcom/sonic-platform-modules-ruijie/common/lib/eepromutil/fantlv.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+class FantlvException(Exception):
+    def __init__(self,  message='fantlverror', code=-100):
+        err = 'errcode: {0} message:{1}'.format(code, message)
+        Exception.__init__(self, err)
+        self.code = code
+        self.message = message
+
+class fan_tlv(object):
+    HEAD_INFO = "\x01\x7e\x01\xf1"
+    VERSION = 0x01                       # E2PROM文件定义的版本号，初始版本为0x01
+    FLAG = 0x7E                       # 新版本E2PROM标识为0x7E
+    HW_VER = 0X01                       # 由主版本号与修正版本两部份构成
+    TYPE = 0xf1                       # 硬件类型定义信息
+    TLV_LEN = 00                        # 有效数据长度（16bit）
+    _FAN_TLV_HDR_LEN = 6
+    _FAN_TLV_CRC_LEN = 2
+
+    _FAN_TLV_TYPE_NAME = 0x02
+    _FAN_TLV_TYPE_SN = 0x03
+    _FAN_TLV_TYPE_HW_INFO = 0x05
+    _FAN_TLV_TYPE_DEV_TYPE = 0x06
+
+    _fandecodetime = 0
+
+    @property
+    def dstatus(self):
+        return self._dstatus
+
+    @property
+    def typename(self):
+        return self._typename
+
+    @property
+    def typesn(self):
+        return self._typesn
+
+    @property
+    def typehwinfo(self):
+        return self._typehwinfo
+
+    @property
+    def typedevtype(self):
+        return self._typedevtype
+
+    @property
+    def fanbus(self):
+        return self._fanbus
+
+    @property
+    def fanloc(self):
+        return self._fanloc
+
+    @property
+    def fandecodetime(self):
+        return self._fandecodetime
+
+    def __init__(self):
+        self._typename = ""
+        self._typesn = ""
+        self._typehwinfo = ""
+        self._typedevtype = ""
+        self._dstatus = 0
+
+    def strtoarr(self, str):
+        s = []
+        if str is not None:
+            for index in str:
+                s.append(index)
+        return s
+
+    def str_to_hex(self,rest_v):
+        value = 0
+        for index in range(len(rest_v)):
+            value |= ord(rest_v[index]) << ((len(rest_v) - index - 1) * 8)
+        return value
+
+    def hex_to_str(self,s):
+        len_t = len(s)
+        if len_t % 2 != 0:
+            return 0
+        ret = ""
+        for t in range(0, len_t / 2):
+            ret += chr(int(s[2 * t:2 * t + 2], 16))
+        return ret
+
+    def generate_fan_value(self):
+        bin_buffer = [chr(0xff)] * 256
+        bin_buffer[0] = chr(self.VERSION)
+        bin_buffer[1] = chr(self.FLAG)
+        bin_buffer[2] = chr(self.HW_VER)
+        bin_buffer[3] = chr(self.TYPE)
+
+        temp_t = "%08x" % self.typedevtype  # 把devtype先处理下
```

Comment:
this makes difficult for english only speaker to review. can these comments be written in english?

Reply:
done!